setup/gcp/installer/forseti_installer.py

```diff
@@ -0,0 +1,291 @@
+
+        # After deployment
+        self.post_install_instructions(deploy_success,
+                                       deployment_name,
+                                       deployment_tpl_path,
+                                       conf_file_path,
+                                       bucket_name)
+
+    def preflight_checks(self):
+        """Pre-flight checks"""
+        self.check_run_properties()
+        self.branch = infer_version(self.config.advanced_mode)
+        self.project_id, authed_user, is_devshell = get_gcloud_info()
+        verify_gcloud_information(self.project_id,
+                                  authed_user,
+                                  self.config.force_no_cloudshell,
+                                  is_devshell)
+        self.organization_id = lookup_organization(self.project_id)
+        check_billing_enabled(self.project_id, self.organization_id)
+        self.format_gcp_service_acct_id()
+        self.gcp_service_account = create_reuse_service_acct(
+            'gcp_service_account',
+            self.gcp_service_account,
+            self.config.advanced_mode,
+            self.config.dry_run)
+
+    def deploy(self, deploy_tpl_path, conf_file_path, bucket_name):
+        """Deploy Forseti using the deployment template
+
+        Args:
+            deploy_tpl_path (str): Deployment template path
+            conf_file_path (str): Configuration file path
+            bucket_name (str): Name of the GCS bucket
+
+        Returns:
+            bool: Whether or not the deployment was successful
+            str: Deployment name
+        """
+        deployment_name, return_code = create_deployment(
+            self.project_id,
+            self.organization_id,
+            deploy_tpl_path,
+            self.config.template_type,
+            self.config.datetimestamp,
+            self.config.dry_run)
+
+        if not return_code:
+            # If deployed successfully, make sure the VM has been initialized,
+            # copy configuration file, deployment template file and
+            # rule files to the GCS bucket
+            conf_output_path = FORSETI_CONF_PATH.format(
+                bucket_name=bucket_name,
+                template_type=self.config.template_type)
+            copy_file_to_destination(
+                conf_file_path, conf_output_path,
+                is_directory=False, dry_run=self.config.dry_run)
+
+            dpl_tpl_output_path = DEPLOYMENT_TEMPLATE_OUTPUT_PATH.format(
+                bucket_name)
+            copy_file_to_destination(
+                deploy_tpl_path, dpl_tpl_output_path,
+                is_directory=False, dry_run=self.config.dry_run)
+
+        return not return_code, deployment_name
+
+    def wait_until_vm_initialized(self, vm_name):
+        """Check vm init status.
+
+        Args:
+            vm_name (str): Name of the VM instance.
+        """
+        installer_type = self.config.template_type.capitalize()
```

Comment:
Hmmm, why would I get a `installer_type` from calling `capitalize()` method?  Can the naming be improved somewhere so that I don't have to jump around to read the code to understand?

Reply:
Renamed template_type to installer_type. capitalize is a built in python string method to capitalize the string.